crates/runtime/src/dataconnector.rs

```diff
@@ -278,3 +305,114 @@ pub async fn get_data(
 
     Ok((table_provider.schema(), batches))
 }
+
+pub trait ListingTableConnector: DataConnector {
+    fn as_any(&self) -> &dyn Any;
+
+    fn get_object_store_url(&self, dataset: &Dataset) -> AnyErrorResult<Url>;
```

Comment:
```suggestion
    fn get_object_store_url(&self, dataset: &Dataset) -> DataConnectorResult<Url>;
```

These should return DataConnectorResult

Reply:
Here and for `get_file_format_and_extension` and `get_csv_format`

Reply:
`get_object_store_url` should return `AnyErrorResult` cause there we return errors which are defined for specific DataConnector
`get_file_format_and_extension`, `get_csv_format` changed to `DataConnectorResult`